invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/Home/components/MostLoanedDocuments/MostLoanedDocuments.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { generatePath } from 'react-router';
+import { Loader, Error } from '../../../../../common/components';
+import { itemSearchQueryUrl, FrontSiteURLS } from '../../../../../common/urls';
+import { Button, Card, Divider } from 'semantic-ui-react';
+import { DocumentCard } from '../DocumentCard';
+import { shortenString } from '../../../../../common/helpers';
+
+export default class MostLoanedDocuments extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchMostLoanedDocuments = props.fetchMostLoanedDocuments;
+    this.showDetailsUrl = FrontSiteURLS.bookDetails;
+    this.maxItemsToDisplay = props.maxDisplayedItems;
+  }
+
+  componentDidMount() {
+    this.fetchMostLoanedDocuments();
+  }
+
+  _showDetailsHandler = document_pid =>
+    this.props.history.push(
+      generatePath(this.showDetailsUrl, { documentPid: document_pid })
```

Comment:
I see that you are using in the container of this component the `withRouter` HOC but I think that is not needed. Could you test that the `generatePath` function works without the previous mentioned HOC and if yes remove the HOC? I think we are a bit overusing this withRouter and is not always necessary:)